test/e2e/deploy_test.go

```diff
@@ -0,0 +1,126 @@
+
+		By("Check Kubernetes/MongoCLI version\n")
+		session = cli.Execute("kubectl", "version")
+		Eventually(session).Should(Say(K8sVersion))
+		session = cli.Execute("mongocli", "--version")
+		Eventually(session).Should(gexec.Exit(0))
+
+		By("Apply All-in-one configuration\n in ")
+		session = cli.Execute("kubectl", "apply", "-f", ConfigAll)
+		Eventually(session.Wait()).Should(Say("customresourcedefinition.apiextensions.k8s.io/atlasclusters.atlas.mongodb.com"))
+		Eventually(
+			cli.GetPodStatus(namespaceOperator),
+			"5m", "3s",
+		).Should(Equal("Running"))
+
+		By("Create secret")
+
+		session = cli.Execute("kubectl", "create", "secret", "generic", "my-atlas-key",
+			"--from-literal=orgId="+os.Getenv("MCLI_ORG_ID"),
+			"--from-literal=publicApiKey="+os.Getenv("MCLI_PUBLIC_API_KEY"),
+			"--from-literal=privateApiKey="+os.Getenv("MCLI_PRIVATE_API_KEY"),
+			"-n", namespaceUserResources,
+		)
+		Eventually(session.Wait()).Should(Say("my-atlas-key created"))
+
+		By("Create Sample Project\n")
+		session = cli.Execute("kubectl", "apply", "-f", ProjectSample, "-n", namespaceUserResources)
+		Eventually(session.Wait()).Should(Say("atlasproject.atlas.mongodb.com/my-project"))
+
+		By("Sample Cluster\n")
+		session = cli.Execute("kubectl", "apply", "-f", ClusterSample, "-n", namespaceUserResources)
+		Eventually(session.Wait()).Should(Say("atlascluster-sample created"))
+
+		By("Wait creating and check that it was created")
+		Eventually(cli.GetGeneration(namespaceUserResources)).Should(Equal("1"))
+		Eventually(
+			cli.IsProjectExist(userProjectConfig.Spec.Name),
+			"7m", "10s",
+		).Should(BeTrue())
+
+		projectID := cli.GetProjectID(userProjectConfig.Spec.Name)
+		GinkgoWriter.Write([]byte("projectID = " + projectID))
+
+		Eventually(
+			cli.IsClusterExist(projectID, userClusterConfig.Spec.Name),
+			"5m", "1s",
+		).Should(BeTrue())
+
+		Eventually(
+			cli.GetClusterStatus(projectID, userClusterConfig.Spec.Name),
+			"35m", "1m",
+		).Should(Equal("IDLE"))
+
+		By("check cluster Attribute") // TODO ...
+		cluster := cli.GetClustersInfo(projectID, userClusterConfig.Spec.Name)
+		Expect(
+			cluster.ProviderSettings.InstanceSizeName,
+		).Should(Equal(userClusterConfig.Spec.ProviderSettings.InstanceSizeName))
+		Expect(
+			cluster.ProviderSettings.ProviderName,
+		).Should(Equal(userClusterConfig.Spec.ProviderSettings.ProviderName))
+		Expect(
+			cluster.ProviderSettings.RegionName,
+		).Should(Equal(userClusterConfig.Spec.ProviderSettings.RegionName))
+
+		By("Update cluster\n")
+		session = cli.Execute("kubectl", "apply", "-f", "data/updated_atlascluster_basic.yaml", "-n", namespaceUserResources) // TODO param
+		Eventually(session.Wait()).Should(Say("atlascluster-sample configured"))
+
+		By("Wait creation")
+		userClusterConfig = cli.LoadUserClusterConfig("data/updated_atlascluster_basic.yaml")
+		Eventually(cli.GetGeneration(namespaceUserResources)).Should(Equal("2"))
```

Comment:
The same as for the create logic the update the code should wait for two things:
1. the observed generation is equal to the generation field (it's ok to check for "2" directly as well)
2. the "status.conditions.Ready : true"